src/edge.cpp

```diff
@@ -0,0 +1,24 @@
+//
+// Created by Renjie Wu on 2019-02-24.
+//
+#include "../include/edge.h"
+namespace pipe_graph{
+    double Edge::edge_id() const {
+        return edge_id_;
```

Comment:
Singe line functions can be defined in header. Compilers are more likely to inline those operations.